lib/code_corps/task/service.ex

```diff
@@ -38,27 +39,31 @@ defmodule CodeCorps.Task.Service do
     {:error, :github}
   end
 
-  @preloads [[github_repo: :github_app_installation], :user]
+  @preloads [:github_issue, [github_repo: :github_app_installation], :user]
 
-  @spec connect_to_github(Task.t) :: {:ok, Task.t} :: {:error, GitHub.api_error_struct}
-  defp connect_to_github(%Task{github_repo_id: nil} = task), do: {:ok, task}
-  defp connect_to_github(%Task{github_repo_id: _} = task) do
-    with {:ok, issue} <- task |> Repo.preload(@preloads) |> GitHub.Issue.create do
-      task |> link_with_github_changeset(issue) |> Repo.update
+  @spec create_on_github(Task.t) :: {:ok, Task.t} :: {:error, GitHub.api_error_struct}
+  defp create_on_github(%Task{github_repo_id: nil} = task), do: {:ok, task}
+  defp create_on_github(%Task{github_repo: _} = task) do
+    with %Task{github_repo: github_repo} = task <- task |> Repo.preload(@preloads),
+         {:ok, payload} <- GitHub.Issue.create(task),
+         {:ok, %GithubIssue{} = github_issue } <- IssueLinker.create_or_update_issue(github_repo, payload) do
```

Comment:
These could be two distinct transaction steps in the function higher up. Just because it doesn't perform a database action, doesn't mean it can't be a step and there might be less branching paths that way.

Steps could be something like `:create_on_github` and `:create_on_codecorps` and then `marshall_result` could deal with a  github api error and a changeset error instead of the `with` clause.

I think it would make the code easier to read kind of like what we did with all the webhook handlers.